catalog/to-consul/consul_node_services_client_ent_test.go

```diff
@@ -0,0 +1,343 @@
+// +build enterprise
+
+package catalog
+
+import (
+	"testing"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+// Test the Consul 1.7 client against Consul Enterprise.
+func TestConsulOnePointSevenClient_NodeServicesWithTag(t *testing.T) {
+	t.Parallel()
+	cases := map[string]struct {
+		ConsulServices []api.CatalogRegistration
+		Exp            []ConsulService
+	}{
+		"no services": {
+			ConsulServices: nil,
+			Exp:            nil,
+		},
+		"no services on k8s node": {
+			ConsulServices: []api.CatalogRegistration{
+				{
+					Node:    "not-k8s",
+					Address: "127.0.0.1",
+					Service: &api.AgentService{
+						ID:      "svc-id",
+						Service: "svc",
+					},
+				},
+			},
+			Exp: nil,
+		},
+		"service with k8s tag on different node": {
+			ConsulServices: []api.CatalogRegistration{
+				{
+					Node:    "not-k8s",
+					Address: "127.0.0.1",
+					Service: &api.AgentService{
+						ID:      "svc-id",
+						Service: "svc",
+						Tags:    []string{"k8s"},
+					},
+				},
+			},
+			Exp: nil,
+		},
+		"service on k8s node but without tag": {
```

Comment:
It would be good to have a test like this, only with a tag or two that don't match "k8s". That will make sure we don't accidentally delete services from a different sync process that's using a different tag.